### ISSUE TITLE
afterSign hook is triggered before code signing happens

### ISSUE THREAD

**Issue by davej** [electron-userland/electron-builder#3504]
* **Version**:  20.37.0

* **Target**: Mac

Config option looks like:
```js
afterSign() {
  console.log("after sign handler called");
}
```

Console output looks like this.

```
after sign handler called
  • signing         file=/{snipped}/dist/mac/TestApp.app identityName=Developer ID Application: Dave Jeffery (SN1PP3D) identityHash=446B886736EC179EF88B902E0515A166202A5FFF provisioningProfile=none
```

**Comment by Kilian** [comment]
Still the case in 20.41.0

**Comment by Kilian** [comment]
It's probably because the call here: https://github.com/electron-userland/electron-builder/blob/ceafebbfda6c68dfb1da85879fd635ad1aae7f25/packages/app-builder-lib/src/platformPackager.ts#L250 

links to a function that resolves immediately: https://github.com/electron-userland/electron-builder/blob/ceafebbfda6c68dfb1da85879fd635ad1aae7f25/packages/app-builder-lib/src/platformPackager.ts#L332-L334

whereas the signing function for mac seems to be called just 'sign', and not 'signApp':
https://github.com/electron-userland/electron-builder/blob/ceafebbfda6c68dfb1da85879fd635ad1aae7f25/packages/app-builder-lib/src/macPackager.ts#L121

What happens is that in the following code, it checks for non-MAS, or target of more than one. If you're building just a DMG, then MAS will be false but it will also be creating a ZIP, setting the target count to two.
https://github.com/electron-userland/electron-builder/blob/09726952278c9977f81a530a10e9f87dcbfcc756/packages/app-builder-lib/src/macPackager.ts#L86-L91

In this function, the entire doPack function is called first (that function includes the `afterSign` call and the `await this.signApp` that immediately resolves) and only after that function is resolved, calls the local `sign` function.

It seems to me the macPackager file needs to be updated: the sign function needs to be renamed to signApp so it's correctly called in the doPack function, and `.then(() => this.sign(appPath, null, null))` on line 89 should be removed.

**Comment by Kilian** [comment]
This ticket can be closed, now that #3912 is merged 🎉
